### PR TITLE
Add an `isApiCompatible()` convenience `Semver` function

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: src/main/java/org/semver4j/Semver.java [hunk 1]
```diff
@@ -515,14 +535,14 @@ public String toString() {
     }
 
     /**
-     * The types of diffs between two versions.
+     * The types of diffs between two versions. The higher the ordinal value of the enum is, the greater is the diff.
```
Comment (piotrooo) [inline src/main/java/org/semver4j/Semver.java:538]: Here should be added an information to keep enums in the order.

File: src/test/java/org/semver4j/SemverTest.java [hunk 2]
```diff
@@ -436,6 +436,18 @@ void shouldSortVersions() {
                 .containsExactly("1.2.0", "1.2.2-beta.1", "1.2.2-rc1", "1.2.2-rc2", "1.2.2", "1.2.3-rc1", "1.2.3-rc2", "1.2.3-rc3", "1.2.3");
     }
 
+    @Test
+    void shouldCheckIsApiCompatible() {
```
Comment (piotrooo) [inline src/test/java/org/semver4j/SemverTest.java:440]: This test method should looks like:

```
@ParameterizedTest
@MethodSource("apiCompatible")
void shouldCheckIsApiCompatible(String version, boolean expected) {
    //given
    Semver semver = new Semver("1.2.3");

    //when
    boolean apiCompatible = semver.isApiCompatible(version);

    //then
    assertThat(apiCompatible).isEqualTo(expected);
}

static Stream<Arguments> apiCompatible() {
    return Stream.of(
            arguments("0.4.3", false),
            arguments("1.1.9", true),
            arguments("1.2.0", true),
            arguments("1.2.3", true),
            arguments("1.2.4", true),
            arguments("1.4.2", true),
            arguments("2.0.0", false),
            arguments("2.2.3", false)
    );
}
```

**Review by piotrooo — CHANGES_REQUESTED** [review]

File: src/main/java/org/semver4j/Semver.java (hunk 1 above)
Comment (sschuberth) [reply src/main/java/org/semver4j/Semver.java:538]: You mean beyond the sentence that I added? Isn't it clear from that already?

File: src/test/java/org/semver4j/SemverTest.java (hunk 2 above)
Comment (sschuberth) [reply src/test/java/org/semver4j/SemverTest.java]: Done.

File: src/main/java/org/semver4j/Semver.java (hunk 1 above)
Comment (piotrooo) [reply src/main/java/org/semver4j/Semver.java:538]: Maybe it is, but I'm worry when someone change the order. But maybe added test will be enough.

**Review by piotrooo — APPROVED** [review]

**Comment by piotrooo** [comment]
Merged - many thanks!

**Pull request merged**